### PR TITLE
Display app name in the registration email title

### PR DESCRIPTION
**Pull request by SunnyWan59**
I changed the title of the registation email from "Welcome to Gitea" to "Welcome to (appname)". I did this by substituting 'Gitea' in register_notify to %s and formating the registration title in mail.go. I changed this for all the languages I could, but I struggled to changed the translation file for Farsi. I have attached the screenshots of before and after and the related issue below. 

Related issue:
https://github.com/go-gitea/gitea/issues/31108

Before change:
Screenshot from 2024-07-04 22-32-17

After:
Screenshot from 2024-07-04 22-32-01

### CODE REVIEW

**Review by delvh — APPROVED** [review]
LGTM once you remove the other changes

**Comment by SunnyWan59** [comment]
> LGTM once you remove the other changes

To be clear remove the changes to the other locales right?

File: services/mailer/mail.go [hunk 1]
```diff
@@ -158,7 +158,8 @@ func SendRegisterNotifyMail(u *user_model.User) {
 		return
 	}
 
-	msg := NewMessage(u.Email, locale.TrString("mail.register_notify"), content.String())
+	appname := setting.AppName
+	msg := NewMessage(u.Email, locale.TrString("mail.register_notify", appname), content.String())
```
Comment (wxiaoguang) [inline services/mailer/mail.go]: The default value of `AppName` is not "Gitea" but "Gitea: Git with a cup of tea", I am not sure whether it is the right value for email title.

Comment (delvh) [reply services/mailer/mail.go]: I also thought about it, but it is most likely customized anyway, so probably not that big of a concern.

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -477,8 +477,8 @@ activate_email = Verify your email address
 activate_email.title = %s, please verify your email address
 activate_email.text = Please click the following link to verify your email address within <b>%s</b>:
 
-register_notify = Welcome to Gitea
-register_notify.title = %[1]s, welcome to %[2]s
+register_notify = Welcome to %s
+register_notify.title = %[1]s, welcome to %[2]
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
register_notify.title = %[1]s, welcome to %[2]s
```

**Review by 6543 — APPROVED** [review]

**Pull request merged**